src/containers/AdminCampaignEdit/sections/CampaignContactsForm/components/ConfigureColumnMappingDialog.tsx

```diff
@@ -99,15 +99,15 @@ const ConfigureColumnMappingDialog: React.FC<ConfigureColumnMappingDialogProps>
   };
 
   useEffect(() => {
-    if (open && !parseComplete) {
+    if (open && !parseComplete && contactsFile) {
       Papa.parse(contactsFile, {
         header: true,
         preview: 2,
         complete: (results: ParseResult<any>) => {
           const columnMappings = results?.meta?.fields?.map(
             (header: string) => ({
               column: header.trim(),
-              remap: namedFields[header.trim()] ?? null
+              remap: (namedFields as any)[header.trim()] ?? null
```

Comment:
really prefer not using any and rather specify correct types upsteam, but if this is urgent, then it's ok

Reply:
Not so urgent that I can't slip in a better type fix! Typescript is already automatically picking up the object structure but seems to really want any here for the property indexing 🤔. Do you know what type might work better?

![image](https://github.com/politics-rewired/Spoke/assets/76596635/63ed6895-efbb-49e7-9ad2-baa14502782d)

Reply:
```
type namedFieldsKeys =
  | "first_name"
  | "firstname"
  | "firstName"
  | "last_name"
  | "lastname"
  | "lastName"
  | "cell"
  | "zip"
  | "external_id"
  | "externalId";
```
This is not ideal, but it's a bit better than casting as any
` const trimmedHeader = header?.trim() as namedFieldsKeys;`

Reply:
not related to what you just did, but trying to add more specific typing. Do you know if this typing is true where it has to be the string "First Name"
```
const namedFields: {
  first_name: typeof firstName;
  firstname: typeof firstName;
  firstName: typeof firstName;
  last_name: typeof lastName;
  lastname: typeof lastName;
  lastName: typeof lastName;
  cell: string;
  zip: string;
  external_id: typeof externalId;
  externalId: typeof externalId;
} = {
  first_name: firstName,
  firstname: firstName,
  firstName,
  last_name: lastName,
  lastname: lastName,
  lastName,
  cell,
  zip,
  external_id: externalId,
  externalId
};
```

Reply:
ok sorry this is actually a better way to do things where I think you no longer need to cast 
```
type fieldAliasKeys = keyof typeof fieldAliases;

type namedFieldsKeys = keyof typeof namedFields;

export type ColumnMapping = {
  column: namedFieldsKeys;
  remap: fieldAliasKeys | null;
};
```

Reply:
🔥 thx!